Sources/Rownd/Models/Context/Auth.swift

```diff
@@ -108,7 +112,7 @@ struct TokenResource: APIResource {
 }
 
 class Auth {
-    static func refreshToken(refreshToken: String, withCompletion completion: @escaping (AuthState?) -> Void) -> Void {
+    static func refreshToken(refreshToken: String?, id_token: String?, withCompletion completion: @escaping (AuthState?) -> Void) -> Void {
```

Comment:
Good re-use of this function! Let's rename this to be a bit more generic like `func fetchToken()`

Then, rather than using two token arguments here, we can overload this function in the following way.

The first function overload would look something like this and handle the refresh token case:

```swift
static func fetchToken(refreshToken: String, withCompletion completion: @escaping (AuthState?) -> Void) -> Void {
  let tokenRequest = TokenRequest(refreshToken: refreshToken)
  return fetchToken(tokenRequest: tokenRequest, withCompletion: completion)
}
```

The second function overload would be something like this, handling the Apple ID token case:

```swift
static func fetchToken(idToken: String, withCompletion completion: @escaping (AuthState?) -> Void) -> Void {
  let tokenRequest = TokenRequest(idToken: idToken, appId: store.state.appConfig.id)
  return fetchToken(tokenRequest: tokenRequest, withCompletion: completion)
}
```


Now, we can pack all of our shared business logic into the final function overload:

```swift
static func fetchToken(tokenRequest: TokenRequest, withCompletion completion: @escaping (AuthState?) -> Void) -> Void {
    var resource = TokenResource()
    resource.headers = [
        "Content-Type": "application/json"
    ]
    let request = APIRequest(resource: resource)
    let encoder = JSONEncoder()
    var body: Data?
    do {
      body = try encoder.encode(tokenRequest)
    } catch {
      return completion(nil)
    }

    request.execute(...)
}
```

So, then the Apple ID code can call the function expecting an ID token, the refresh flow can call this function with that type of token. This makes the internal API contract more explicit and easier to test.